torchinductor/ir.py

```diff
@@ -201,7 +201,13 @@ def is_triton(device):
 
 
 class IRNode(object):
```

Comment:
Could we just put everything on the IRNode base class?  Having things scattered on specific node constructors doesn't seem great.

Something like:
```
class IRNode:
   _current_origins = set()
   @staticmethod
   @contexlib.contextmanager
   def current_origin(origins: Set[fx.Node])
      old = IRNode._current_origins
      IRNode._current_origins = old | origins
      yield
      IRNode._current_origins = old

   def __init__(self):
     self.origins = set(self._current_origins)
```


Then we could just throw a:
```
with IRNode.current_origins({n}):
   ...
```

Around the entire body of `run_node(n)`.

Reply:
Yeah I thought about it, but I wasn't sure it was right for all IRNode to have this property on it. Is there any case where a subclass of IRNode having an `origins` is nonsensical?

Reply:
No, every node can have an origins.